### DISCUSSION TITLE
French Response. Let's simplify them drastically

### DISCUSSION THREAD

**Discussion by jlpouffier** [OHF-Voice/intents#1354] [Supported Sentences]
Hello 👋 

This discussion is a case to simplify the responses of our French intent massively.
I decided to write that post in English because I believe it could be interesting for other language leaders if they want to do something similar.

# Context
Today, the French responses are very long and very specific.

- If I ask to open the cover of the living room, the answer will be "Openings of the living room open"
- If I ask to turn on a specific light, the answer will be "`name_of_the_speficif_light` turned on"
- If I ask to set the brightness of a specific light to 47%, the answer will be "Brightness of `name_of_the_speficif_light` set to 47%"
- etc etc

As I said: Long and specific.

# Issues
## Issue 1: Our responses are badly cached

As explained in the [chapter 2 in Year of the Voice](https://www.home-assistant.io/blog/2023/04/27/year-of-the-voice-chapter-2/), Home Assistant is doing a good job caching the voice response to be re-used as much as possible.

This is particularly true for the few of us that have decided to use a local TTS component in their pipeline (Piper).

If the same text is output twice for two different queries, the second TTS is free, because it is already cached.

**Conclusion: The more specific our responses are, the less cached they can be, the longer the response time is, the more expensive the "CPU cost" is (Wherever it is: Locally or in Nabu Casa cloud)** 

## Issue 2: Our responses slows us down

Our French intents, and by extension almost every language, are in an early stage of the development process.
This is a phase where we want to be able to contribute fast in order to support as much functionality as possible.

I would welcome without any issue the complexity of our responses if the project was mature, but as it is not the case, I am questioning its usefulness _right now_
Today, supporting a new functionality means having to deal with the complexity of its answers, and most of us are trying to stick to the conventions already in place in the project.

For example, we have different answers if we query 
- an entity 
- an entity in an area
- all entities of an area
- all entities of all areas

**Conclusion: The more specific our responses are, the more complex our project is becoming, and the slower we become**

## Issue 3: Our responses are too specific but not mature enough, so we are creating some very unnatural responses sometimes

For example, today if you ask about the state of a température sensor, the answer will be `the state of the device living room temperature is 23.04 °C` 
Which may be the most unnatural way of answering this question.
This is because we are specific.... but not specific enough. 
It's like we opened a Pandora box trying to be specific, and we struggle to close it now.

And other commun example of lack of maturity is the fact that we are failing to detect masculine vs feminine form. So if I ask to turn off lights, I'll get as a reply `lumières éteins` which makes it unnatural.

**Conclusion: Being specific will always raise weird exceptions where the answers feel unnatural... And this only way to fix it is even more specificity**

# Proposed solution 

We drastically simplify all our responses, so that
- They are cached better
- We reduce complexity and increase our own velocity
- We limit the unnatural response, at the cost of more direct responses

Concrete proposal
- Every **action** _(Intent where we ask Home Assistant to perform an action: Open a cover, set a brightness, turn off a light, start a vacuum, make some coffee, whatever)_: The same response. An acknowledgment such as `OK` or `D'accord`. Note, this is the "Alexa's way" ... even they, being way more mature than us, decided not to care about specifying their answers.
  - Turn off the bedroom light > `OK`
  - Open all the covers in the living room > `OK`
- Every intent that needs to return a state: We return the state. And that's it. 
  - What's the humidity of the bedroom > `47%`
  - Give me the bedroom's temperature > `23.7 °C`
 - Every Yes/No question will be replied with either Yes or No. And that's it.  _(Today we try to add a state for example, but we are failing to translate them correctly. For example, if I ask `La machine a café est allumée ?`, I get `Non, off`_ )
   - Is the coffee maker on? > `No`

I'll ping the few of us that are contributing to the French intents, but ultimately, I am waiting for an answer from our language leader.

Thx!

JLo

**Comment by jlpouffier** [comment]
Ping a few french contributors 

- @slegars56 
- @piitaya 
- @bemble 
- @CedricFinance 

And of course pinging the language leader 
- @flexy2dd

**Comment by slegars56** [comment]
+1, 
responses sound a bit weird: "Lumières éteint". Moreover, the French model in Piper (Siwis-medium) struggles with some sounds like "éteint"(etɛ̃).

BTW, I noticed inconsistencies with Whisper (small-int8) outputs: for example, the output "Quelles fenêtres sont ouvertes" is sometimes returned with "Quels fenêtres sont ouvertes".
It requires taking care of more cases with the intents to get a maximum of matches (quel[l][e][s] with this example). 
I have no clue about the impact on performances.

**Reply by jlpouffier** [reply]
> +1, 
> responses sound a bit weird: "Lumières éteint". Moreover, the French model in Piper (Siwis-medium) struggles with some sounds like "éteint"(etɛ̃).
> 
> BTW, I noticed inconsistencies with Whisper (small-int8) outputs: for example, the output "Quelles fenêtres sont ouvertes" is sometimes returned with "Quels fenêtres sont ouvertes".
> It requires taking care of more cases with the intents to get a maximum of matches (quel[l][e][s] with this example). 
> I have no clue about the impact on performances.
> 
> 

Thx for your example. I am adding the case "failing to detect masculine and feminine form" under the "issue 3"

**Comment by bemble** [comment]
My opinion is, if we just want an acknowledgment, then `éteint` or `ouvert` should be enough. I fully agree with your points. When the system will be more mature, I wish one day we would be able to choose between a TTS response or a sound and the TTS response could also be a random feedback between `D'accord`, `OK` or `C'est fait !`.

With that said, as the service is brand new, I can understand that people do not trust it yet, so a full acknowledgment can help to trust the service, moreover when you don't have a visual feedback on your old phone. The ideal solution would be to have a `verbose mode`, which could be enabled by default for now, and when disabled replies short acknowledgments.

**Reply by jlpouffier** [reply]
Interesting point about the trust issue.


Honestly with hours of testing and tweaking intents, I never got a case where Home Assistant mistakenly took the wrong action.


Either it works.. or does not at all.
I'll be curious to research that at some point.


To come back to your other points:


About the "slightly more specific acknowledgement":
I would challenge that if we really want to move fast.
Even a simple verb as you proposed create so much complexity. Some of which unsupported by HA itself


- éteint or éteinte ? That is not even per domain... It's mer entity
-  how many verbs? 
  - Éteint/allumé for light 
  - Ouvert/Fermé for cover
  - how about switches ? 
    - I have a coffee maker switch... So allumé/éteins seams fine
    - but I also have a switch for the AdGuardHome filtering... In this case Activé/Désactivé should be more natural
  - Vacuum ? Démarré/Arrêté ?


All of this is solved by a simpler acknowledgement like "OK" ... At the cost of a slightly more direct and less meaningful answer.


About the two mode verbose or not: That seems to be leading toward more complexity. At least now 
Unless we find a very clever idea to switch modes simply ... I am not sure I see this idea viable on the current state of the project.

Do you have a cleaver idea ? 🤗

**Reply by bemble** [reply]
I trust the service, but I can understand users need to play a little with it to trust it.

To me, a great solution would be to reply with a random `D'accord`, `OK` or `C'est fait !` because I don't like too much verbosity in the replies. If it were just me, I'd like just a sound (and a status code to display a light accordingly), I hate when my Homepod replies "Ok Pierre", worst experience is when I run "Hey Siri, good night", and it replies "OK Pierre, I turned on the scene successfuly". But it's just my own taste.

But we have to keep in mind that's just the beginning (and what a beginning); it's not fine tunning phase yet. The "verbose mode" has to be implemented in the core; it's not something we can do with the intent unless you provide a kind of `fr-FR-quiet` variation, which does not fit with the intent rules.

The topic addressed is very subjective. The choice to give short replies should not be language-specific, to keep the answers consistent across languages and so, pipelines.  Remember there's [this guide](https://developers.home-assistant.io/docs/voice/intent-recognition/style-guide), and yes it's written to be concise, but not that concise.

French and probably other languages raised this subject because it's more verbose than English and bring other problem like some verbs/words that do not exist (or concisely), matching the gender of the entities etc. Simplification might be the right answer, but we cannot decide by ourselves the direction the replies take for all languages. I think @balloob and @synesthesiam have to be part of that discussion as to me, it's not French-related.

**Reply by jlpouffier** [reply]
Very interesting guide ! 

The french responses violate these two rules:

- don't repeat the name of the entity in the response
- don't repeat the name of the area or its entities in the response

 I am fine taking a decision across language. No problem with me ;)

**Reply by synesthesiam** [reply]
In the future, we will consider different categories of responses such as "short" and "long". I'd argue for a boolean-style response too -- basically an acknowledging "grunt" from your voice assistant :smile:

**Reply by bemble** [reply]
So, we will make French compliant with the current language rules (you know, french people and rules... 😅) and wait for the short/long categories.

**Reply by flexy2dd** [reply]
I kind of agree with @bemble that having several answers "humanises" the experience
but I understand that one is forced to respect the law of the majority, or at least the system standard  (and it's complicated for a french people 😅) 

At the moment, as @bemble said, the system is new and we have to deal with all the other languages, 
so in my opinion we should keep this proposal "sous le coude" because I find it interesting and see if other languages are in the same situation and interested

**Reply by HarvsG** [reply]
> In the future, we will consider different categories of responses such as "short" and "long". I'd argue for a boolean-style response too -- basically an acknowledging "grunt" from your voice assistant 😄

Particularly as we are now context aware, if (1) the command and the device are in the same area and (2) the change in device state is likely immediately apparent (e.g light on, blind closed) (negative e.g lock locked or relay turned on) then a "Grunt/Chime/Done" exclamation should be more than enough.
